rust/kernel/allocator.rs

```diff
@@ -2,29 +2,136 @@
 
 //! Allocator support.
 
+use core::alloc::AllocError;
 use core::alloc::{GlobalAlloc, Layout};
 use core::ptr;
+use core::ptr::NonNull;
 
 use crate::bindings;
+use crate::static_assert;
 
-struct KernelAllocator;
+pub(crate) struct KernelAllocator;
+
+fn is_power_of_two(x: usize) -> bool {
+    (x != 0) && ((x & (x - 1)) == 0)
```

Comment:
There is a builtin function for this: https://doc.rust-lang.org/std/primitive.usize.html#method.is_power_of_two